### PR TITLE
Remove backend packages

### PR DESCRIPTION
**Pull request by duckontheweb**
**Related Issue(s):** 

- #432 

**Description:**

- **Removes backend-specific code, configuration, and documentation**
  Going forward, the SQLAlchemy and PgSTAC backends will be distributed as separate packages and managed in their own repos ([`stac-fastapi-sqlalchemy`](https://github.com/stac-utils/stac-fastapi-sqlalchemy) and [`stac-fastapi-pgstac`](https://github.com/stac-utils/stac-fastapi-pgstac), respectively). The main `Dockerfile` and `docker-compose.yml` have been removed since there is no longer a deployable backend associated with this repo.
- **Distributes `stac-fastapi.api`, `stac-fastapi.types`, and `stac-fastapi.extensions` as modules under a single `stac-fastapi` package.**
   Import statements for the individual modules will not change, but downstream applications will need to update their dependency from something like...
   ```
   stac-fastapi.api==2.4.1
   stac-fastapi.types==2.4.1
   stac-fastapi.extensions==2.4.1
   ```
   ...to something like...
   ```
   stac-fastapi==2.5.0
   ```

**PR Checklist:**

- [x] Code is formatted and linted (run `pre-commit run --all-files`)
- [x] Tests pass (run `make test`)
- [x] Documentation has been updated to reflect changes, if applicable, and docs build successfully (run `make docs`)
- [x] Changes are added to the [CHANGELOG](https://github.com/stac-utils/stac-fastapi/blob/master/CHANGES.md).
- [ ] Remove or add CI to implement validation checks

### CODE REVIEW

**Review by geospatial-jeff — COMMENTED** [review]
A few more comments, typing them out here because there isn't a great place to comment in the diff.
- Test command in the makfile should run inside the docker container.
- The test data at `tests/data/joplin` probably isn't used anywhere, but maybe its nice to keep around?
- Can remove the `alembic` exclude from the `find_namespace_packages` call in `setup.py`.

I'm starting on the sqlalchemy backend today, if I find anything else I'll add here.

File: docker-compose.docs.yml [hunk 1]
```diff
@@ -7,12 +7,5 @@ services:
       context: .
```
[the remark this answers is not in the document]
Comment (duckontheweb) [reply docker-compose.docs.yml]: Fixed in 25f0c61f704e0aca59386129a8e31f4b31726803

File: scripts/publish [hunk 2]
```diff
@@ -61,12 +52,9 @@ fi
 if [ "${BASH_SOURCE[0]}" = "${0}" ]; then
     for PACKAGE_DIR in "${SUBPACKAGE_DIRS[@]}"
```
[the remark this answers is not in the document]
Comment (duckontheweb) [reply scripts/publish]: This should be fixed by 735307b3a1a24d6f8bb8888db74ec0162e377aa7

**Comment by duckontheweb** [comment]
> * The test data at `tests/data/joplin` probably isn't used anywhere, but maybe its nice to keep around?

Yeah, it might make sense to keep it around for now. I'm thinking that as we expand the test suite we may want  a simple backend mocker that just reads files from a directory. Having these files available would probably make that easier to put together in the future.

**Comment by duckontheweb** [comment]
> * Can remove the `alembic` exclude from the `find_namespace_packages` call in `setup.py`.

Fixed in 2d72c588151fc16eec1564553a653a039bce6c58

**Comment by gadomski** [comment]
NB there have been some changes to backend-specific code that were dropped by 22d6a01, so we will need to go back and backport any of those to the backend-specific repos (if they haven't been backported already).

**Comment by gadomski** [comment]
Since this appears to have stalled out, I've made an RFC with a plan to finish the backend breakout. Tl;dr: I'm proposing we punt on the separation of responsibilities (e.g. https://github.com/stac-utils/stac-fastapi/pull/453) -- while it's a good idea, it's becoming a blocker that's holding the **stac-fastapi** ecosystem in an undefined state. While we might end up with pain down the road to synchronize the backend repos with the main ones, it's my hope that we can use API breakages v3.0 to solve the separation of concerns in the medium-term.

**Comment by geospatial-jeff** [comment]
@gadomski That makes sense to me, we can always come back to these changes in the future in a later major version.  I think the more important thing is getting the backends split out, regardless of the separation of concerns.

**Pull request closed without merging**